src/lib/autodiscovery/autodiscovery_task_executor.py

```diff
@@ -0,0 +1,108 @@
+import asyncio
+from datetime import datetime, timedelta
+from typing import Dict, List, Optional
+
+from lib.autodiscovery.autodiscovery_manager import AutodiscoveryManager
+from lib.configuration import config
+from lib.context import LoggingContext
+from lib.metrics import AutodiscoveryGCPService, GCPService
+
+AUTODISCOVERY_QUERY_INTERVAL_SEC = config.get_autodiscovery_querry_interval() * 60
```

Comment:
Besides this var not being used, isn't the autodiscovery also depending on the main loop's query interval (inside run_docker's loop)? Maybe not, just asking.